### PR TITLE
iOS & Android integration tests

### PR DESCRIPTION
**Pull request by radex**
Changed:

- Copied over / set up again the integration test harness from Purple
- Added missing Metro/Babel/ESLint configuration + dependencies necessary to package watermelon through metro for integration tests and for all checks to pass
- Added `yarn test:ios` and `yarn test:android`
- Hooked up test:ios into Travis CI
- Moved BridgeTestReporter from the main Watermelon bundle to the integration test harness apps (🍉 clients don't need that)
- Moved `__tests__` inside src
- Wrote up contribution guidelines for native code

Known issues:

- Android tests don't run on CI yet
- I couldn't figure out how to make Metro understand our absolute paths without setting up rn-cli.js, transformer, and a bunch of package.jsons (Metro finds those files using Haste). I'll make a separate issue for this

### CODE REVIEW

**Comment by radex** [comment]
@rozPierog Check out this branch. `yarn test:android` works fine on my machine, but we need to hook it up to Travis

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,5 +42,31 @@ We recommend VS Code with ESLint, Flow, and Prettier (with prettier-eslint enabl
...
+### Integration tests
+
+If you change native bridge code or `adapter/sqlite` code, it's recommended to run integration tests that run the entire Watermelon code with SQLite and React Native in the loop:
+
+```bash
+yarn test:ios
+yarn test:android
+```
+
+You can also run iOS integration tests manually by opening the `native/iosTest/WatermelonTester.xcworkspace` project and hitting Cmd+U.
```
Comment (rozPierog) [inline CONTRIBUTING.md]: Giving a recommendation on how to run it in XCode but not in Android Studio. Should I add line about Android?

Comment (radex) [reply CONTRIBUTING.md]: Yeah!

File: package.json [hunk 2]
```diff
@@ -10,7 +10,9 @@
     "flow": "flow check",
     "eslint": "eslint ./src -c ./.eslintrc.yml --cache --cache-location ./.cache/.eslintcache",
     "test": "jest --config=./jest.config.js --forceExit --detectOpenHandles --no-cache",
-    "ci:check": "concurrently 'yarn test' 'yarn eslint' 'yarn flow' --kill-others-on-fail"
+    "ci:check": "concurrently 'npm run test' 'npm run eslint' 'npm run flow' --kill-others-on-fail",
```
Comment (kokusGr) [inline package.json]: What is the reason for the change?

**Comment by kokusGr** [comment]
CR 👌 Only one question.

File: package.json (hunk 2 above)
Comment (radex) [reply package.json]: saves a little bit of time on Travis CI, because we don't have to install yarn, and I don't think it has any negative effects

Comment (kokusGr) [reply package.json]: Ok, cool :)

**Comment by radex** [comment]
@rozPierog can you update the docs as mentioned above? Or we can merge this and you could update docs on master @kokusGr

**Comment by rozPierog** [comment]
@radex could you resolve conflicts?

**Comment by radex** [comment]
@kokusGr @rozPierog fixed

**Comment by kokusGr** [comment]
Everything looks good :)
